authors/apps/articles/views.py

```diff
@@ -82,4 +93,72 @@ def put(cls, request, slug):
     def delete(cls, request, slug):
         article = get_object_or_404(Article, slug=slug, delete_status=False)
         article.delete()
-        return Response({"messege": "article deleted successfully"}, status=status.HTTP_200_OK)
\ No newline at end of file
+        return Response({"messege": "article deleted successfully"}, status=status.HTTP_200_OK)
+
+class FavoriteArticleCreate(generics.ListCreateAPIView):
+    """If the user feels satisfied with the article, he can favourite it """
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    queryset = FavoriteArticle.objects.all()
+    serializer_class = FavoriteArticleSerializer
+    article_serializer_class = ArticleSerializer
+
+    def check_article_exists(self, **kwargs):
+        """
+        Check whether article exists
+        """
+        article = get_object_or_404(
+            Article, 
+            slug=kwargs.get('slug')
+        )
+        return article
+
+    def post(self, request, **kwargs):
+        """
+        Favorite an article
+        """
+        article = self.check_article_exists(**kwargs)
+        favorite = FavoriteArticle.objects.filter(
+            favorited_by=request.user, 
+            article=article
+        )
+        if not favorite:
+            serializer = self.serializer_class(data={'favorited':True})
+            serializer.is_valid(raise_exception=True)
+            serializer.save(
+                article=article,
+                favorited_by=request.user
+            )
+            return Response(
+                {"message": "You have successfully favorited this article "},
+                status=status.HTTP_200_OK
+            )
+        return Response(
+            {"message": "You already favorited this article"},
+            status=status.HTTP_200_OK
+        )
+
+
+class UnFavoriteArticleDestroy(generics.DestroyAPIView):
+    """If he feels disatisfied with the article he can Unfavourite it. """
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    queryset = FavoriteArticle.objects.all()
+    serializer_class = FavoriteArticleSerializer
+    def delete(self, request, **kwargs):
+        """
+        Unfavorite an article
+        """
+        queryset = FavoriteArticle.objects.filter(favorited_by=request.user)
+        favorited = queryset.filter(article_id=(Article.objects.filter(slug=kwargs.get('slug')).first()).id).first()
+        serializer = FavoriteArticleSerializer(favorited)
+
+        if not favorited:
+            return Response(
+                {
+                    'message': "You have already unfavourited this article"
+                }, status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Hello @nadralia , great work and logic, however, for a user who has never `favorited` an article, a proper error message is needed: instead of `'message': "You have already unfavourited this article"`, let the message be close to `"message":"You have not favourited this article"`.

Reply:
@nadralia, Besides this, the other logic in your code looks good to me

Reply:
Fixed it